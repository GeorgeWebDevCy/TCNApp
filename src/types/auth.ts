export type AuthMethod = 'password' | 'pin' | 'biometric';

export type AccountType =
  | 'member'
  | 'vendor'
  | 'staff'
  | 'admin'
  | 'guest'
  | (string & {});

export type AccountStatus =
  | 'active'
  | 'pending'
  | 'rejected'
  | 'suspended'
  | (string & {});

export type RegisterAccountType = 'member' | 'vendor';

export interface MembershipBenefit {
  id: string;
  title: string;
  description?: string;
  discountPercentage?: number;
}

export interface MembershipInfo {
  tier: string;
  expiresAt: string | null;
  benefits: MembershipBenefit[];
}

export interface MemberQrCode {
  token: string;
  payload?: string | null;
  issuedAt?: string | null;
  expiresAt?: string | null;
}

export interface MembershipPlan {
  id: string;
  name: string;
  price: number;
  currency: string;
  interval?: 'day' | 'week' | 'month' | 'year';
  description?: string;
  features?: string[];
  highlight?: boolean;
  metadata?: Record<string, unknown>;
}

export interface WooCommerceCredentialBundle {
  consumerKey: string;
  consumerSecret: string;
  basicAuthorizationHeader?: string | null;
}

export interface AuthUser {
  id: number;
  email: string;
  name: string;
  firstName?: string | null;
  lastName?: string | null;
  avatarUrl?: string;
  membership?: MembershipInfo | null;
  woocommerceCredentials?: WooCommerceCredentialBundle | null;
  accountType?: AccountType | null;
  accountStatus?: AccountStatus | null;
  vendorTier?: string | null;
  vendorStatus?: AccountStatus | null;
  qrPayload?: string | null;
  qrToken?: string | null;
}

export interface AuthState {
  isAuthenticated: boolean;
  isLocked: boolean;
  isLoading: boolean;
  user: AuthUser | null;
  membership: MembershipInfo | null;
  memberQrCode: MemberQrCode | null;
  authMethod: AuthMethod | null;
  error: string | null;
  hasPasswordAuthenticated: boolean;
}

export interface LoginOptions {
  email: string;
  password: string;
  mode?: 'token';
  remember?: boolean;
}

export interface PinLoginOptions {
  pin: string;
}

export interface RegisterOptions {
  username: string;
  email: string;
  password: string;
  firstName?: string;
  lastName?: string;
  registrationDate?: string;
  accountType?: RegisterAccountType;
}

export interface ResetPasswordOptions {
  identifier: string;
  verificationCode: string;
  newPassword: string;
  resetKey?: string;
}

export interface AuthContextValue {
  state: AuthState;
  loginWithPassword: (options: LoginOptions) => Promise<void>;
  loginWithPin: (options: PinLoginOptions) => Promise<void>;
  loginWithBiometrics: (promptMessage?: string) => Promise<void>;
  changePassword: (options: {
    currentPassword: string;
    newPassword: string;
  }) => Promise<void>;
  registerPin: (pin: string) => Promise<void>;
  removePin: () => Promise<void>;
  logout: () => Promise<void>;
  resetError: () => void;
  refreshSession: () => Promise<void>;
  getSessionToken: () => Promise<string | null>;
  requestPasswordReset: (identifier: string) => Promise<string | undefined>;
  registerAccount: (options: RegisterOptions) => Promise<string | undefined>;
  resetPasswordWithCode: (
    options: ResetPasswordOptions,
  ) => Promise<string | undefined>;
  updateProfileAvatar: (options: {
    uri: string;
    fileName?: string;
    mimeType?: string;
  }) => Promise<AuthUser>;
  deleteProfileAvatar: () => Promise<AuthUser>;
}

export interface MemberValidationResult {
  token: string;
  valid: boolean;
  memberName?: string | null;
  membershipTier?: string | null;
  allowedDiscount?: number | null;
  membership?: MembershipInfo | null;
  message?: string | null;
}
