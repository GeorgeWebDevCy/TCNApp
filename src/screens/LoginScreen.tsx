import React, { useCallback, useEffect, useMemo, useState } from 'react';
import {
  Alert,
  Pressable,
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import { BiometricLoginButton } from '../components/BiometricLoginButton';
import { ForgotPasswordModal } from '../components/ForgotPasswordModal';
import { LanguageSwitcher } from '../components/LanguageSwitcher';
import { LoginHeader } from '../components/LoginHeader';
import { PinLoginForm } from '../components/PinLoginForm';
import { RegisterModal } from '../components/RegisterModal';
import { WordPressLoginForm } from '../components/WordPressLoginForm';
import { useAuthContext } from '../contexts/AuthContext';
import { useLocalization } from '../contexts/LocalizationContext';
import { useAuthAvailability } from '../hooks/useAuthAvailability';
import { RegisterOptions } from '../types/auth';
import { COLORS } from '../config/theme';

type AuthTabId = 'password' | 'pin';

export const LoginScreen: React.FC = () => {
  const {
    state,
    loginWithPassword,
    loginWithPin,
    loginWithBiometrics,
    registerPin,
    removePin,
    resetError,
    requestPasswordReset,
    registerAccount,
  } = useAuthContext();
  const {
    pin: hasStoredPin,
    biometrics: biometricsEnabled,
    biometricsSupported,
    biometryType,
    loading: availabilityLoading,
    refresh,
  } = useAuthAvailability();
  const { t, translateError } = useLocalization();
  const [activeTab, setActiveTab] = useState<AuthTabId>('password');
  const [lastAttempt, setLastAttempt] = useState<
    'password' | 'pin' | 'biometric' | null
  >(null);
  const [pinError, setPinError] = useState<string | null>(null);
  const [isForgotPasswordVisible, setForgotPasswordVisible] = useState(false);
  const [isRegisterVisible, setRegisterVisible] = useState(false);

  const authTabs = useMemo(() => {
    const tabs: Array<{ id: AuthTabId; label: string }> = [
      { id: 'password', label: t('login.tabs.password') },
    ];

    if (hasStoredPin) {
      tabs.push({ id: 'pin', label: t('login.tabs.pin') });
    }

    return tabs;
  }, [hasStoredPin, t]);

  useEffect(() => {
    if (hasStoredPin && state.isLocked) {
      setActiveTab('pin');
    }
  }, [hasStoredPin, state.isLocked]);

  useEffect(() => {
    if (!hasStoredPin && activeTab === 'pin') {
      setActiveTab('password');
    }
  }, [activeTab, hasStoredPin]);

  const handlePasswordSubmit = useCallback(
    async ({ username, password }: { username: string; password: string }) => {
      setLastAttempt('password');
      resetError();
      await loginWithPassword({ username, password });
    },
    [loginWithPassword, resetError],
  );

  const handlePinSubmit = useCallback(
    async (pin: string) => {
      setLastAttempt('pin');
      setPinError(null);
      resetError();
      await loginWithPin({ pin });
    },
    [loginWithPin, resetError],
  );

  const handlePinCreate = useCallback(
    async (pin: string) => {
      setLastAttempt('pin');
      try {
        await registerPin(pin);
        await refresh();
        Alert.alert(
          t('login.alerts.pinSaved.title'),
          t('login.alerts.pinSaved.message'),
        );
      } catch (error) {
        const message =
          error instanceof Error ? error.message : 'errors.pinSaveGeneric';
        setPinError(message);
      }
    },
    [refresh, registerPin, t],
  );

  const handleRemovePin = useCallback(async () => {
    try {
      await removePin();
      await refresh();
      Alert.alert(
        t('login.alerts.pinRemoved.title'),
        t('login.alerts.pinRemoved.message'),
      );
    } catch (error) {
      const message =
        error instanceof Error ? error.message : 'errors.pinRemoveGeneric';
      setPinError(message);
    }
  }, [refresh, removePin, t]);

  const handleBiometricLogin = useCallback(async () => {
    setLastAttempt('biometric');
    resetError();
    await loginWithBiometrics(t('biometrics.prompt'));
  }, [loginWithBiometrics, resetError, t]);

  const handleRequestPasswordReset = useCallback(
    (identifier: string) => requestPasswordReset(identifier),
    [requestPasswordReset],
  );

  const handleRegisterAccount = useCallback(
    (options: RegisterOptions) => registerAccount(options),
    [registerAccount],
  );

  const changeTab = useCallback(
    (tabId: (typeof authTabs)[number]['id']) => {
      setActiveTab(tabId);
      setLastAttempt(null);
      setPinError(null);
      resetError();
    },
    [resetError],
  );

  const activeError = useMemo(() => {
    if (lastAttempt === 'pin') {
      return pinError ?? state.error;
    }
    if (lastAttempt === 'password') {
      return state.error;
    }
    if (lastAttempt === 'biometric') {
      return state.error;
    }
    return activeTab === 'pin' ? pinError ?? state.error : state.error;
  }, [activeTab, lastAttempt, pinError, state.error]);

  const isLoading = state.isLoading;
  const greeting =
    state.user?.name !== undefined
      ? t('login.greeting', { replace: { name: state.user.name } })
      : undefined;
  const lockMessage = state.isLocked ? t('login.lockMessage') : undefined;
  const translatedError = translateError(activeError);
  const passwordError =
    lastAttempt === 'password' ? translateError(state.error) : null;

  return (
    <SafeAreaView style={styles.safeArea}>
      <StatusBar barStyle="dark-content" backgroundColor={COLORS.surface} />
      <ScrollView
        contentContainerStyle={styles.scrollContainer}
        keyboardShouldPersistTaps="handled"
      >
        <View style={styles.switcherContainer}>
          <LanguageSwitcher />
        </View>
        <View style={styles.card}>
          <LoginHeader subtitle={greeting ?? t('login.subtitle')} />

          {lockMessage ? (
            <Text style={styles.lockMessage}>{lockMessage}</Text>
          ) : null}

          <View style={styles.tabRow}>
            {authTabs.map(tab => {
              const isActive = tab.id === activeTab;
              return (
                <Pressable
                  key={tab.id}
                  onPress={() => changeTab(tab.id)}
                  accessibilityRole="button"
                  style={[styles.tabButton, isActive && styles.tabButtonActive]}
                >
                  <Text
                    style={
                      isActive
                        ? styles.tabButtonTextActive
                        : styles.tabButtonText
                    }
                  >
                    {tab.label}
                  </Text>
                </Pressable>
              );
            })}
          </View>

          {activeTab === 'password' ? (
            <WordPressLoginForm
              loading={isLoading && lastAttempt === 'password'}
              error={passwordError}
              onSubmit={handlePasswordSubmit}
              onForgotPassword={() => setForgotPasswordVisible(true)}
              onRegister={() => setRegisterVisible(true)}
            />
          ) : hasStoredPin ? (
            <PinLoginForm
              hasPin={hasStoredPin}
              canManagePin={state.hasPasswordAuthenticated}
              loading={isLoading && lastAttempt === 'pin'}
              error={translatedError}
              onSubmit={handlePinSubmit}
              onCreatePin={handlePinCreate}
              onResetPin={hasStoredPin ? handleRemovePin : undefined}
            />
          ) : (
            <View style={styles.noticeBox}>
              <Text style={styles.noticeText}>
                {t('login.prompts.pinSetup')}
              </Text>
            </View>
          )}

          {biometricsEnabled ? (
            <>
              <View style={styles.dividerSection}>
                <View style={styles.divider} />
                <Text style={styles.dividerText}>{t('common.or')}</Text>
                <View style={styles.divider} />
              </View>

              <BiometricLoginButton
                available={!availabilityLoading}
                biometryType={biometryType}
                loading={isLoading && lastAttempt === 'biometric'}
                onPress={handleBiometricLogin}
              />
            </>
          ) : biometricsSupported ? (
            <Text style={styles.noticeText}>
              {t('login.prompts.biometricSetup')}
            </Text>
          ) : null}
        </View>
      </ScrollView>
      <ForgotPasswordModal
        visible={isForgotPasswordVisible}
        onClose={() => setForgotPasswordVisible(false)}
        onSubmit={handleRequestPasswordReset}
      />
      <RegisterModal
        visible={isRegisterVisible}
        onClose={() => setRegisterVisible(false)}
        onSubmit={handleRegisterAccount}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
    backgroundColor: COLORS.surface,
  },
  scrollContainer: {
    flexGrow: 1,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 24,
    backgroundColor: COLORS.surfaceMuted,
  },
  switcherContainer: {
    width: '100%',
    maxWidth: 420,
    alignItems: 'flex-end',
    marginBottom: 16,
  },
  card: {
    width: '100%',
    maxWidth: 420,
    padding: 24,
    borderRadius: 24,
    backgroundColor: COLORS.surface,
    shadowColor: COLORS.textPrimary,
    shadowOpacity: 0.08,
    shadowOffset: { width: 0, height: 12 },
    shadowRadius: 24,
    elevation: 8,
    gap: 24,
  },
  tabRow: {
    flexDirection: 'row',
    gap: 12,
    backgroundColor: COLORS.background,
    padding: 4,
    borderRadius: 999,
  },
  tabButton: {
    flex: 1,
    borderRadius: 999,
    backgroundColor: 'transparent',
  },
  tabButtonActive: {
    backgroundColor: COLORS.surface,
    shadowOpacity: 0.09,
    shadowRadius: 8,
    shadowOffset: { width: 0, height: 2 },
    elevation: 2,
  },
  tabButtonText: {
    textAlign: 'center',
    paddingVertical: 10,
    fontWeight: '600',
    color: COLORS.textSecondary,
  },
  tabButtonTextActive: {
    textAlign: 'center',
    paddingVertical: 10,
    fontWeight: '700',
    color: COLORS.textPrimary,
  },
  dividerSection: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 12,
  },
  divider: {
    flex: 1,
    height: 1,
    backgroundColor: COLORS.border,
  },
  dividerText: {
    color: COLORS.textSecondary,
    fontWeight: '600',
  },
  noticeBox: {
    marginTop: 8,
    padding: 16,
    borderWidth: 1,
    borderColor: COLORS.border,
    borderRadius: 16,
    backgroundColor: COLORS.surfaceMuted,
  },
  noticeText: {
    textAlign: 'center',
    color: COLORS.textSecondary,
    fontSize: 14,
  },
  lockMessage: {
    textAlign: 'center',
    color: COLORS.infoText,
    fontWeight: '600',
  },
});
