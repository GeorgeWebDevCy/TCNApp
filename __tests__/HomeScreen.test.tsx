import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { Text } from 'react-native';
import { HomeScreen, getMaxDiscount } from '../src/screens/HomeScreen';
import { MembershipInfo } from '../src/types/auth';

const membership: MembershipInfo = {
  tier: 'Gold',
  expiresAt: '2030-01-10T00:00:00.000Z',
  benefits: [
    {
      id: 'travel',
      title: 'Travel deals',
      description: '10% off flights',
      discountPercentage: 10,
    },
    {
      id: 'dining',
      title: 'Dining rewards',
      description: '15% off restaurants',
      discountPercentage: 15,
    },
  ],
};

jest.mock('../src/contexts/AuthContext', () => ({
  useAuthContext: () => ({
    state: {
      isAuthenticated: true,
      isLocked: false,
      isLoading: false,
      user: {
        id: 1,
        email: 'jane@example.com',
        name: 'Jane Doe',
        firstName: 'Jane',
        lastName: 'Doe',
        membership,
      },
      membership,
      authMethod: 'password',
      error: null,
      hasPasswordAuthenticated: true,
    },
    logout: jest.fn(),
  }),
}));

jest.mock('../src/contexts/TransactionContext', () => ({
  useTransactionContext: () => ({
    transactions: [
      {
        id: 'tx-1',
        memberToken: 'member-token',
        memberName: 'Jane Doe',
        vendorName: 'Vendor Plaza',
        status: 'completed',
        discountPercentage: 10,
        discountAmount: 120,
        netAmount: 1080,
        grossAmount: 1200,
        createdAt: '2024-01-01T00:00:00.000Z',
      },
    ],
    addTransaction: jest.fn(),
    replaceTransaction: jest.fn(),
    patchTransaction: jest.fn(),
  }),
}));

jest.mock('../src/contexts/LocalizationContext', () => {
  const { translations } = jest.requireActual(
    '../src/localization/translations',
  );

  const resolve = (key: string): unknown => {
    return key.split('.').reduce<unknown>((current, segment) => {
      if (current && typeof current === 'object') {
        return (current as Record<string, unknown>)[segment];
      }
      return undefined;
    }, translations.en);
  };

  const format = (
    template: string,
    replacements?: Record<string, string | number>,
  ) => {
    if (!replacements) {
      return template;
    }

    return template.replace(/{{\s*([^\s{}]+)\s*}}/g, (match, token) => {
      const value = replacements[token];
      return value !== undefined && value !== null ? String(value) : match;
    });
  };

  return {
    useLocalization: () => ({
      language: 'en',
      setLanguage: jest.fn().mockResolvedValue(undefined),
      translateError: (value: string | null) => value,
      t: (
        key: string,
        options?: { replace?: Record<string, string | number> },
      ) => {
        const value = resolve(key);
        if (typeof value === 'string') {
          return format(value, options?.replace);
        }
        return key;
      },
    }),
  };
});

jest.mock('../src/notifications/OneSignalProvider', () => ({
  useOneSignalNotifications: () => ({
    preferences: { marketing: true, reminders: true },
    updatePreference: jest.fn(),
    activeNotification: null,
    activeNotificationOrigin: null,
    clearActiveNotification: jest.fn(),
    pendingNavigationTarget: null,
    consumeNavigationTarget: jest.fn(),
  }),
}));

describe('HomeScreen', () => {
  it('renders the membership overview', () => {
    let renderer: ReactTestRenderer.ReactTestRenderer | undefined;
    const onManageProfile = jest.fn();
    ReactTestRenderer.act(() => {
      renderer = ReactTestRenderer.create(
        <HomeScreen onManageProfile={onManageProfile} />,
      );
    });

    const textNodes = renderer
      ?.root.findAllByType(Text)
      .map(node => node.props.children)
      .flat();

    expect(textNodes).toContain('Welcome, Jane Doe!');
    expect(textNodes).toContain('Recent savings');
    expect(textNodes).toContain('Vendor Plaza');
  });
});

describe('getMaxDiscount', () => {
  it('returns the maximum discount from the benefit list', () => {
    expect(getMaxDiscount(membership.benefits)).toBe(15);
  });

  it('returns null when no numeric discounts are present', () => {
    expect(
      getMaxDiscount([
        { id: 'a', title: 'Perk', description: 'Bonus points' },
        { id: 'b', title: 'Another', discountPercentage: Number.NaN },
      ]),
    ).toBeNull();
  });
});
