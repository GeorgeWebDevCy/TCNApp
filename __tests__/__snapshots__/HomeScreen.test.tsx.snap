// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders the membership overview 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "flexGrow": 1,
        "gap": 20,
        "padding": 24,
      }
    }
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          {
            "alignSelf": "flex-end",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "flex-end",
              "gap": 4,
            }
          }
        >
          <Text
            style={
              {
                "color": "#475569",
                "fontSize": 12,
                "fontWeight": "600",
              }
            }
          >
            Language
          </Text>
          <View
            accessibilityLabel="Language"
            accessibilityRole="radiogroup"
            style={
              {
                "backgroundColor": "#E2E8F0",
                "borderRadius": 999,
                "flexDirection": "row",
                "gap": 8,
                "padding": 4,
              }
            }
          >
            <View
              accessibilityLabel="Switch to English"
              accessibilityRole="radio"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": true,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={8}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderRadius": 20,
                    "height": 40,
                    "justifyContent": "center",
                    "width": 40,
                  },
                  {
                    "backgroundColor": "#FFFFFF",
                    "elevation": 2,
                    "shadowColor": "#0F172A",
                    "shadowOffset": {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 6,
                  },
                ]
              }
            >
              <Text
                style={
                  {
                    "fontSize": 22,
                  }
                }
              >
                ðŸ‡ºðŸ‡¸
              </Text>
            </View>
            <View
              accessibilityLabel="Switch to Thai"
              accessibilityRole="radio"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": false,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={8}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "borderRadius": 20,
                    "height": 40,
                    "justifyContent": "center",
                    "width": 40,
                  },
                  false,
                ]
              }
            >
              <Text
                style={
                  {
                    "fontSize": 22,
                  }
                }
              >
                ðŸ‡¹ðŸ‡­
              </Text>
            </View>
          </View>
        </View>
      </View>
      <Text
        style={
          {
            "color": "#0F172A",
            "fontSize": 24,
            "fontWeight": "700",
          }
        }
      >
        Welcome, Jane Doe!
      </Text>
      <Text
        style={
          {
            "color": "#475569",
            "fontSize": 16,
          }
        }
      >
        jane@example.com
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-start",
            "backgroundColor": "#E0F2FE",
            "borderColor": "#38BDF8",
            "borderRadius": 999,
            "borderWidth": 1,
            "marginTop": 12,
            "paddingHorizontal": 18,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#0369A1",
              "fontSize": 14,
              "fontWeight": "600",
            }
          }
        >
          Manage profile
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#EEF2FF",
            "borderRadius": 16,
            "gap": 12,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "#312E81",
              "fontSize": 18,
              "fontWeight": "700",
            }
          }
        >
          Membership status
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              {
                "color": "#4338CA",
                "fontSize": 16,
                "fontWeight": "600",
              }
            }
          >
            Tier
          </Text>
          <Text
            style={
              {
                "color": "#1E1B4B",
                "fontSize": 20,
                "fontWeight": "700",
              }
            }
          >
            Gold
          </Text>
        </View>
        <Text
          style={
            {
              "color": "#4C1D95",
              "fontSize": 14,
            }
          }
        >
          Renews on Jan 10, 2030
        </Text>
        <Text
          style={
            {
              "color": "#312E81",
              "fontSize": 14,
            }
          }
        >
          Save up to 15% with member perks
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#F8FAFC",
            "borderRadius": 16,
            "gap": 12,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "#0F172A",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Your benefits
        </Text>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "gap": 12,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#2563EB",
                "borderRadius": 4,
                "height": 8,
                "marginTop": 6,
                "width": 8,
              }
            }
          />
          <View
            style={
              {
                "flex": 1,
                "gap": 4,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1F2937",
                  "fontSize": 16,
                  "fontWeight": "600",
                }
              }
            >
              Travel deals
            </Text>
            <Text
              style={
                {
                  "color": "#4B5563",
                  "fontSize": 14,
                }
              }
            >
              10% off flights
            </Text>
            <Text
              style={
                {
                  "color": "#059669",
                  "fontSize": 14,
                  "fontWeight": "600",
                }
              }
            >
              -10%
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "gap": 12,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#2563EB",
                "borderRadius": 4,
                "height": 8,
                "marginTop": 6,
                "width": 8,
              }
            }
          />
          <View
            style={
              {
                "flex": 1,
                "gap": 4,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1F2937",
                  "fontSize": 16,
                  "fontWeight": "600",
                }
              }
            >
              Dining rewards
            </Text>
            <Text
              style={
                {
                  "color": "#4B5563",
                  "fontSize": 14,
                }
              }
            >
              15% off restaurants
            </Text>
            <Text
              style={
                {
                  "color": "#059669",
                  "fontSize": 14,
                  "fontWeight": "600",
                }
              }
            >
              -15%
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#F8FAFC",
            "borderColor": "#E2E8F0",
            "borderRadius": 16,
            "borderWidth": 1,
            "gap": 16,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "#1E293B",
              "fontSize": 18,
              "fontWeight": "600",
              "marginBottom": 12,
            }
          }
        >
          Notifications
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 16,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              {
                "flex": 1,
                "gap": 4,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#0F172A",
                  "fontSize": 16,
                  "fontWeight": "600",
                }
              }
            >
              Marketing push alerts
            </Text>
            <Text
              style={
                {
                  "color": "#475569",
                  "fontSize": 14,
                }
              }
            >
              Get personalized promotions from partner vendors.
            </Text>
          </View>
          <RCTSwitch
            accessibilityLabel="Marketing push alerts"
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignSelf": "flex-start",
              }
            }
            value={true}
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 16,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              {
                "flex": 1,
                "gap": 4,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#0F172A",
                  "fontSize": 16,
                  "fontWeight": "600",
                }
              }
            >
              Renewal reminders
            </Text>
            <Text
              style={
                {
                  "color": "#475569",
                  "fontSize": 14,
                }
              }
            >
              Receive reminders before your membership expires.
            </Text>
          </View>
          <RCTSwitch
            accessibilityLabel="Renewal reminders"
            accessibilityRole="switch"
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignSelf": "flex-start",
              }
            }
            value={true}
          />
        </View>
      </View>
      <View
        style={
          {
            "gap": 12,
          }
        }
      >
        <Text
          style={
            {
              "color": "#1E293B",
              "fontSize": 18,
              "fontWeight": "600",
              "marginBottom": 12,
            }
          }
        >
          Quick actions
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "gap": 12,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#F1F5F9",
                "borderColor": "#CBD5F5",
                "borderRadius": 10,
                "borderWidth": 1,
                "paddingHorizontal": 16,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1D4ED8",
                  "fontWeight": "600",
                }
              }
            >
              View vendors
            </Text>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#F1F5F9",
                "borderColor": "#CBD5F5",
                "borderRadius": 10,
                "borderWidth": 1,
                "paddingHorizontal": 16,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1D4ED8",
                  "fontWeight": "600",
                }
              }
            >
              Upgrade membership
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "center",
            "backgroundColor": "#2563EB",
            "borderRadius": 12,
            "marginTop": 24,
            "paddingHorizontal": 24,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Log out
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
